tenseal/tensors/ckksvector.cpp

```diff
@@ -176,13 +198,23 @@ CKKSVector& CKKSVector::mul_inplace(CKKSVector to_mul) {
         throw invalid_argument("can't multiply vectors of different sizes");
     }
 
+    if (this->context->is_auto_mod_switch() &&
+        this->ciphertext.parms_id() != to_mul.ciphertext.parms_id()) {
+        set_to_same_mod(this->context, this->ciphertext, to_mul.ciphertext);
+    }
+
     this->context->evaluator->multiply_inplace(this->ciphertext,
                                                to_mul.ciphertext);
 
-    // TODO: can make this optional
-    // relineraize after ciphertext-ciphertext multiplication
-    this->context->evaluator->relinearize_inplace(this->ciphertext,
-                                                  this->context->relin_keys());
+    if (this->context->is_auto_relin()) {
+        this->context->evaluator->relinearize_inplace(
+            this->ciphertext, this->context->relin_keys());
+    }
+
+    if (this->context->is_auto_rescale()) {
+        this->context->evaluator->rescale_to_next_inplace(this->ciphertext);
```

Comment:
Shouldn't we check if the ciphertext got to last_parms_id()? It will throw an exception when it gets there.

Is there a test to capture the situation?

Reply:
we leave it to SEAL to throw an error when needed, but yeah, there is no test for this, I will add one